### ISSUE TITLE
Build vsphere-clone failing with "host '' not found" when specifying cluster and resource pool on DRS enabled cluster

### ISSUE THREAD

**Issue by MasterWaldo** [hashicorp/packer#10069]
When I try cloning a VM the build errors out almost immediately with the error "host '' not found". 
The vSphere cluster I'm running against has DRS enabled. According to the docs, I should not need to specify host. 
I am Admin on the target resource pool, and am able to use Terraform to clone a VM off of the template using all the same values for cluster, resource pool, network, etc.

Using vSphere 6.7.0u3, DRS is enabled on the cluster
Using Packer 1.6.4 (also tried 1.5.2, 1.6.0, 1.6.1)

#### Reproduction Steps

Using the following config, run `packer build`  (some values masked for privacy)

### Simplified Packer Buildfile

```
source "vsphere-clone" "vsphere"{
  vcenter_server = "vc1.***.com"
  username = ***
  password = ***
  insecure_connection = true
  datacenter = "DC-1"
}

build{
  source "sources.vsphere-clone.vsphere"{
    cluster = "CLS-1"
    resource_pool = "Pool1"

    network = "Network-Public"
    template = "template_RHEL7_6"
    vm_name = "packer_test"
    ssh_username = "root"
  }
}
```
### Log Fragments and crash.log files
```
> ./1.6.4/packer build vsphere_clone.pkr.hcl 
vsphere-clone.vsphere: output will be in this color.

==> vsphere-clone.vsphere: Cloning VM...
Build 'vsphere-clone.vsphere' errored after 559 milliseconds 670 microseconds: host '' not found

==> Wait completed after 559 milliseconds 738 microseconds

==> Some builds didn't complete successfully and had errors:
--> vsphere-clone.vsphere: host '' not found

==> Builds finished but no artifacts were created.
```

Set the env var `PACKER_LOG=1` for maximum log detail.
[detailed_log.txt](https://github.com/hashicorp/packer/files/5351002/detailed_log.txt)

**Comment by SwampDragons** [comment]
Hi, thanks for reaching out. We'll try to take a look at getting this fixed

**Comment by SwampDragons** [comment]
this may be related to #9623

**Comment by mashiutz** [comment]
Hey, i am the one who started the issue mentioned above.
I gotta say that i stil cannot work with packer (have tried 1.6.4 today), been trying to use this in our environment for over a year now.

Same issue, although doing a new build and not a clone - Im using a Cluster with DRS.
While wrting down the cluster all lower-case, i get a "resource pool not found".

**Comment by SwampDragons** [comment]
So @MasterWaldo just to make extra clear -- if you lowercase your cluster name 
`    cluster = "CLS-1" `

**Comment by SwampDragons** [comment]
ah nevermind -- didn't read close enough. It was a typo in the config.

**Comment by SwampDragons** [comment]
Hi again, was just tinkering and I think was able to reproduce -- While you don't need to set a host, you do need to set a [datastore](https://www.packer.io/docs/builders/vmware/vsphere-iso#datastore).  This is true for both vsphere-iso and vsphere-clone builders.

If you set a datastore, can you get your build up and running?

**Comment by MasterWaldo** [comment]
Interesting. If I set datastore, I get a permission denied error
`> pckr build vsphere_clone.pkr.hcl 
vsphere-clone.vsphere: output will be in this color.

==> vsphere-clone.vsphere: Cloning VM...
Build 'vsphere-clone.vsphere' errored: ServerFaultCode: Permission to perform this operation was denied.

==> Some builds didn't complete successfully and had errors:
--> vsphere-clone.vsphere: ServerFaultCode: Permission to perform this operation was denied.

==> Builds finished but no artifacts were created.`

But it looks like I have all of the perms, I think.
[Vmware_Perms.txt](https://github.com/hashicorp/packer/files/5458887/Vmware_Perms.txt)

As mentioned on my original post, I'm able to clone off the target template, on the same cluster, datastore, etc, using Terraform. I would imagine the perms would be similar?

**Comment by SwampDragons** [comment]
oy. Sure would help if VSphere would return _which_ permission was missing, eh?  

I can make a custom build with a bunch of extra logging so we can try to figure out specifically which API call is failing for you; would you be willing to try that out with me?

**Comment by MasterWaldo** [comment]
Absolutely

**Comment by SwampDragons** [comment]
okay, pop one of these out of our CI and let's see what we get when you run with the environment var PACKER_LOG=1 https://app.circleci.com/pipelines/github/hashicorp/packer/7771/workflows/352df52d-6af6-4e57-8dee-1a5a45363cac/jobs/89638/artifacts

**Comment by MasterWaldo** [comment]
Here's the interesting looking part of the debug log. Full log attached as well.
`
2020/10/30 09:54:05 [INFO] (telemetry) Starting builder 
2020/10/30 09:54:05 packer_linux_amd64 plugin: No CD files specified. CD disk will not be made.
==> vsphere-clone.vsphere: Cloning VM...
2020/10/30 09:54:05 packer_linux_amd64 plugin: [DEBUG 10069] about to call template.Clone command
2020/10/30 09:54:05 packer_linux_amd64 plugin: [DEBUG 10069] Inside VirtualMachineDriver.Clone
2020/10/30 09:54:05 packer_linux_amd64 plugin: [DEBUG 10069] inside FindDatastore call; name is NFS-Recap and host is
2020/10/30 09:54:05 packer_linux_amd64 plugin: [DEBUG 10069] Error from calling template.Clone command
2020/10/30 09:54:05 [INFO] (telemetry) ending 
==> Wait completed after 358 milliseconds 944 microseconds
2020/10/30 09:54:05 machine readable: error-count []string{"1"}
==> Some builds didn't complete successfully and had errors:
2020/10/30 09:54:05 machine readable: vsphere-clone.vsphere,error []string{"Error calling vm.vm.Clone task: ServerFaultCode: Permission to perform this operation was denied."}
==> Builds finished but no artifacts were created.
Build 'vsphere-clone.vsphere' errored after 358 milliseconds 886 microseconds: Error calling vm.vm.Clone task: ServerFaultCode: Permission to perform this operation was denied.

==> Wait completed after 358 milliseconds 944 microseconds

==> Some builds didn't complete successfully and had errors:
--> vsphere-clone.vsphere: Error calling vm.vm.Clone task: ServerFaultCode: Permission to perform this operation was denied.

==> Builds finished but no artifacts were created.`

[10069_custom_bin_error.txt](https://github.com/hashicorp/packer/files/5465854/10069_custom_bin_error.txt)

**Comment by SwampDragons** [comment]
Yep, that looks like the important part. The vm.vm.Clone task is defined by the vmware-maintained SDK; As far as I can tell it's just the constructor for the API call that then does the Clone VM task

So I see that your permissions list does contain `Clone virtual machine`. I'm no expert in vSphere permissions but maybe there's some issue with permissions being applied asymmetrically, like your user has permission to clone generally, but not that specific VM or something?

**Comment by MasterWaldo** [comment]
Sorry I haven't updated, working with my admins to try and figure out what we can and can't have permission wise
It looks like I may be missing this permission:
`Datastore -> Low level file operations`
or this one
```
Distributed switch (this object):
Read-only
```
But I'm not sure, because the Terraform vsphere module seems to have a very similar permission requirement and I have no issues with that one :P.

**Comment by MasterWaldo** [comment]
I believe we finally tracked down the issue. We were not specifying `folder`, so the VM was getting cloned into the root level folder of the Datacenter, which we don't have permissions on. As soon as we specified a folder that fell in our resource pool, we were able to successfully clone a machine.
 For some reason I was under the impression that it would default to the resource pool root instead of the datacenter root, not sure why I thought that. Maybe some other tool I used worked that way?
Anyways, I think this can probably be closed, it in the end was user error.
